packages/eslint-plugin/src/rules/typedef.ts

```diff
@@ -60,7 +60,21 @@ export default util.createRule<[Options], MessageIds>({
       [OptionKeys.VariableDeclarationIgnoreFunction]: false,
     },
   ],
-  create(context, [options]) {
+  create(
+    context,
+    [
+      {
+        arrayDestructuring,
```

Comment:
I went with destructuring options to avoid `options[Option.xyz]` everywhere, but I can revert this part if desirable.